frontend/tests/controllers/BadgesTest.php

```diff
@@ -47,7 +112,56 @@ public function testAllBadges() {
                 "$alias:> The file test.json doesn't exist."
             );
 
-            // From here I must run the test.json + query.sql
+            if (file_exists($testPath) && file_exists($queryPath)) {
+                self::cleanDb();
+                FileHandler::SetFileUploader($this->createFileUploaderMock());
+                $content = json_decode(file_get_contents($testPath), true);
+
+                // omegaUp admin user must be created always.
+                $omegaup = UserFactory::createAdminUser(new UserParams([
+                    'username' => 'omegaup_admin',
+                    'password' => 'omegaup_admin',
+                ]));
+
+                foreach ($content['actions'] as $action) {
+                    switch ($action['type']) {
+                        case 'changeTime':
+                            $time = strtotime($action['time']);
+                            Time::setTimeForTesting($time);
+                            break;
+
+                        case 'apicalls':
+                            foreach ($action['apicalls'] as $apicall) {
+                                self::RunRequests($apicall);
```

Comment:
nit:

```suggestion
                                self::RunRequest($apicall);
```

sólo está ejecutando un único request.